samza-core/src/main/java/org/apache/samza/container/RunLoopFactory.java

```diff
@@ -98,7 +98,7 @@ public Boolean apply(TaskInstance t) {
       log.info("Run loop in asynchronous mode.");
 
       return new AsyncRunLoop(
-        JavaConversions.mapAsJavaMap(taskInstances),
+        JavaConverters.mapAsJavaMapConverter(taskInstances).asJava(),
```

Comment:
I think `JavaConverters#mapAsJavaMap` etc. should work for conversions like these in Java code. If true, let's change all other occurrences too.